src/msg/async/rdma/RDMAStack.cc

```diff
@@ -291,6 +291,7 @@ void RDMADispatcher::polling()
         for (auto& qp: dead_qps) {
           perf_logger->dec(l_msgr_rdma_active_queue_pair);
           ldout(cct, 10) << __func__ << " finally delete qp = " << qp << dendl;
+          std::lock_guard l{lock}; // protect qp when dealing with it in handle_rx_event
```

Comment:
This does not make sense. handle_rx_event() is invoked from the same thread (basically it was invoked just before we process dead_qps). So there is nothing to protect.

I suppose you are confused by previous "if (!dead_queue_pairs.empty()) {" check (line 284), but this variable is seen outside current event loop, so there we take a lock in order to protect the "dead_queue_pairs", but not QP itself.

Reply:
@rouming 
You're right. I've walked through the code again, there's only one rdma-polling thread(one RDMADispatcher obj)